library/choose_disk.py

```diff
@@ -353,46 +353,69 @@ def expand_disks(lookup_disks, ceph_type="", module=None):
     return final_disks
 
 
-def disk_label(partition):
+def disk_label(partition, current_fsid, ceph_disk):
     '''
     Reports if a partition is containing some ceph structures
     '''
-    # 1) let's search for legacy metadata made by ceph-disk
-    stdout = subprocess.check_output(["blkid", "-s", "PARTLABEL", "-o", "value", "{}".format(partition)])
 
+    # 1) let's search for metadata coming from ceph-volume
+    try:
+        output_cmd = subprocess.Popen(["ceph-volume", "lvm", "list", "--format=json", partition],
+                                      stdout=subprocess.PIPE)
+        raw_json, _ = output_cmd.communicate()
+        try:
+            json_dict = json.loads(raw_json)
+        except (ValueError, TypeError) as e:
+            fatal("Cannot parse ceph-volume properly : {}".format(e))
+
+        # If the disk have some metadata, the json is filled with information
+        for key, value in json_dict.items():
+            # As we parse a partition it should only have a single entry unless that's an error
```

Comment:
this is not correct, we don't force a single partition, why would that be an error?

Reply:
I do force the parsing of the partitio with "output_cmd = subprocess.Popen(["ceph-volume", "lvm", "list", "--format=json", partition"]